botocore/operation.py

```diff
@@ -151,10 +150,8 @@ def request_created(request, **kwargs):
             # a request has already been signed without needing
             # to acquire the lock.
             if not getattr(request, '_is_signed', False):
-                with self._lock:
```

Comment:
I think we still want the lock.  This was to prevent the same request from being signed multiple times because the operation isn't tied to an endpoint/request.  Even now that the signers are thread-safe against signing multiple requests concurrently, I don't think they're thread-safe against signing the same request multiple times, especially with things like body checksums on file like objects.